src/overlays/fbdemos/ovl_fbdemo_wipe3/z_fbdemo_wipe3.c

```diff
@@ -17,37 +19,159 @@ void TransitionWipe3_SetColor(void* thisx, u32 color);
 void TransitionWipe3_SetEnvColor(void* thisx, u32 color);
 s32 TransitionWipe3_IsDone(void* thisx);
 
-#if 0
+#include "assets/overlays/ovl_fbdemo_wipe3/ovl_fbdemo_wipe3.c"
+
+TexturePtr sTransWipe3Textures[] = {
+    fbdemo_tex_000520, fbdemo_tex_000920, fbdemo_tex_000D20, fbdemo_tex_001120,
+    fbdemo_tex_001520, fbdemo_tex_000520, fbdemo_tex_000520, fbdemo_tex_000520,
+};
+
 TransitionInit TransitionWipe3_InitVars = {
-    TransitionWipe3_Init,
-    TransitionWipe3_Destroy,
-    TransitionWipe3_Update,
-    TransitionWipe3_Draw,
-    TransitionWipe3_Start,
-    TransitionWipe3_SetType,
-    TransitionWipe3_SetColor,
-    TransitionWipe3_SetEnvColor,
-    TransitionWipe3_IsDone,
+    TransitionWipe3_Init,     TransitionWipe3_Destroy,     TransitionWipe3_Update,
+    TransitionWipe3_Draw,     TransitionWipe3_Start,       TransitionWipe3_SetType,
+    TransitionWipe3_SetColor, TransitionWipe3_SetEnvColor, TransitionWipe3_IsDone,
 };
 
-#endif
+#define TRANS3_GET_SPEED(type) (type & 1)
+#define TRANS3_GET_TEX_INDEX(type) ((type >> 1) & 7)
+#define TRANS3_GET_COLOR(type) ((type >> 4) & 3)
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_DIR_IN,
+    /* 1 */ TRANS_WIPE3_DIR_OUT
+} TransitionWipe3Direction;
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_SPEED_SLOW,
+    /* 1 */ TRANS_WIPE3_SPEED_FAST
+} TransitionWipe3Speed;
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_COLOR_BLACK,
+    /* 1 */ TRANS_WIPE3_COLOR_WHITE,
+    /* 2 */ TRANS_WIPE3_COLOR_GRAY
+} TransitionWipe3Color;
+
+void TransitionWipe3_Start(void* thisx) {
+    TransitionWipe3* this = THIS;
+
+    this->isDone = false;
+    this->curTexture = sTransWipe3Textures[this->texIndex & 7];
```

Comment:
```suggestion
    this->curTexture = sTransWipe3Textures[this->texIndex % ARRAY_COUNT(sTransWipe3Textures)];
```

Reply:
I remember this not working.

Reply:
Could you check again?

Reply:
![image](https://github.com/zeldaret/mm/assets/35883445/285cba54-7bc0-40fa-b4cd-cab333cf6b3a)
IDK why it mismatches in that way. Usually it just does a `mod` operation instead of an `and`

Reply:
Ohh annoying. What about an `ARRAY_COUNTU`?

Reply:
THAT is seems happy with.